project/CliOptions.scala

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka
+
+case class CliOption[T](value: T)
+
+object CliOption {
+  def apply[T](path: String, default: T)(implicit ev: CliOptionParser[T]): CliOption[T] = ev.parse(path, default)
+
+  implicit class BooleanCliOption(cliOption: CliOption[Boolean]) {
```

Comment:
BTW: an alternative formulation would be using phantom types on a method that lives directly on CliOption :-)